ignore_run_exports_from_substr recipe contains channel spec

ignore_run_exports_from_substr recipe file shouldn't set the channel for the `python` dependency in the recipe itself https://github.com/conda/conda-build/blob/6d9eaf6fefc9bad7f327c8a22841349b7888bf15/tests/test-recipes/metadata/ignore_run_exports_from_substr/meta.yaml#L12

This coincidentally works with `conda-libmamba-solver` but not with `conda-rattler-solver`. Originally discovered through running the test suite with CRS in https://github.com/conda/conda-build/pull/5989:

```
FAILED tests/test_api_build.py::test_recipe_builds[ignore_run_exports_from_substr] - conda_rattler_solver._conda_build_exceptions.ExplainedDependencyNeedsBuildingError: Unsatisfiable dependencies for platform linux-64: {MatchSpec("ncurses=6.6[build=hdb14827_0]")}
Cannot solve the request because of: No candidates were found for ncurses 6.6.* hdb14827_0.
```

Test recipe should be fixed and channel can be passed in the test directly if needed.

--- a/tests/test-recipes/metadata/ignore_run_exports_from_substr/meta.yaml
+++ b/tests/test-recipes/metadata/ignore_run_exports_from_substr/meta.yaml
@@ -9,6 +9,6 @@ build:
 
 requirements:
   host:
-    - conda-forge::python
+    - python
   run:
     - numpy

--- a/tests/test_api_build.py
+++ b/tests/test_api_build.py
@@ -2153,9 +2153,12 @@ def test_api_build_inject_jinja2_vars_on_first_pass(testing_config):
     api.build(recipe_dir, config=testing_config)
 
 
-def test_ignore_run_exports_from_substr(monkeypatch, tmp_path, capsys):
+def test_ignore_run_exports_from_substr(monkeypatch, tmp_path, capsys, testing_config):
     monkeypatch.chdir(tmp_path)
-    api.build(str(metadata_path / "ignore_run_exports_from_substr"))
+    testing_config.channel_urls = ["conda-forge"]
+    api.build(
+        str(metadata_path / "ignore_run_exports_from_substr"), config=testing_config
+    )
     assert "- python_abi " in capsys.readouterr().out
 
 

